scalene/scalene_profiler.py

```diff
@@ -1189,20 +1187,11 @@ def should_trace(filename: str) -> bool:
         if "scalene/scalene" in filename:
             # Don't profile the profiler.
             return False
-        found_in_profile_only = False
-        for prof in profile_only_list:
-            if prof in filename:
-                found_in_profile_only = True
-                break
-
+        found_in_profile_only = any(prof in filename for prof in profile_only_list)
         if not found_in_profile_only:
             return False
```

Comment:
Shouldn't this be
```python
        if profile_only_list and not any(prof in filename for prof in profile_only_list):
            return False

        if Scalene.__args.profile_all:
            # Profile everything else, except for "only" choices.
            return True
```

Reply:
I think the logic should be as follows:
* If `profile-only` is specified, exclude files that don't match any provided pattern.

I am committing a change that I hope simplifies this and makes it clear.